users/serializers.py

```diff
@@ -257,4 +255,57 @@ def create(self, validated_data):
         """
         self.request.user.set_password(validated_data.get("new_password"))
         self.request.user.save()
-        return self.request.user
\ No newline at end of file
+        return self.request.user
+
+
+class TimeLogSerializer(serializers.Serializer):
+    """ timelog serializer for clock-in and clock-out
+        from slack api
+    """
+
+    user = None
+    team_id = serializers.CharField()
+    channel_id = serializers.CharField()
+    channel_name = serializers.CharField()
+    user_id = serializers.CharField()
+    
+    def validate_team_id(self, team_id):
+        """ validate if the team source of the
+            request came from the right team. (swiftkind)
+            if not, deny access.
+        """
+        if team_id != settings.SLACK_TEAM_ID:
+            raise serializers.ValidationError(
+                _('[Invalid Request] Wrong workspace.'),
+                code="invalid_request",
+            )
+
+        return team_id
+
+    def validate_user_id(self, user_id):
+        """ validate if the user is a valid member
+            of the workspace.
+        """
+        self.user = User.objects.filter(slack_id=user_id).first()
+        if not self.user:
+            raise serializers.ValidationError(
+                _('[Invalid Request] Requestor is not a registered user.'),
+                code="invalid_request",
+            )
+        
+        return user_id
+    
+    def create(self, validated_data):
+        """ create or update timelog instance
+        """
+        # get user timelog
+        time_log = TimeLog.objects.filter(user=self.user, time_out=None)
```

Comment:
set `time_in` field to `auto_now_add=True` # this will automatically put timestamp when a timelog obj is created.

```
time_log = TimeLog.objects.get_or_create(user=self.user, time_out=None)
if not created:
    time_log.time_out = timezone.now()

return time_log
```

Reply:
there is a flaw in your code, what if the user enters the command multiple times? your code will create multiple log instance, we won't know which is correct.

You should add a checker to check if the there are existing timelog for that specific date (current date). only 1 log per day should be created